code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm

```diff
@@ -219,12 +219,89 @@
 	var/shock_timer = 0
 	process_flags = ORGANIC | SYNTHETIC
 
-/datum/reagent/teslium/on_mob_life(mob/living/carbon/M)
+/datum/reagent/teslium/on_mob_add(mob/living/carbon/H)
+	var/total_burn	= 0
+	var/total_brute	= 0
+	var/tplus = world.time - H.timeofdeath	//length of time spent dead
+	var/tlimit = 9000
+	var/obj/item/organ/heart = H.getorgan(/obj/item/organ/heart)
+	var/HALFWAYCRITDEATH = ((HEALTH_THRESHOLD_CRIT + HEALTH_THRESHOLD_DEAD) * 0.5)
+
+	if(H.stat == DEAD)
+		H.visible_message("<span class='warning'>[H]'s body convulses a bit.</span>")
+		playsound(src, "bodyfall", 50, 1)
+		playsound(src, 'sound/machines/defib_zap.ogg', 75, 1, -1)
+		total_brute	= H.getBruteLoss()
+		total_burn	= H.getFireLoss()
+		if(isliving(H.pulledby))		//CLEAR!
+			var/mob/living/M = H.pulledby
+			if(M.electrocute_act(30, H))
+				M.visible_message("<span class='danger'>[M] is electrocuted by [M.p_their()] contact with [H]!</span>")
+				M.emote("scream")
+
+
+		var/failed = 0
+
+		if (H.suiciding)
+			failed = 1
```

Comment:
you really should add a `return` after each of the `failed = 1`

Reply:
Yeah like I said it's pretty jank - I was just trying to rig it up to make sure everything was working. I plan to update to clean it up but have been pounding on the bee code the past few hours so just wanted to throw this up.

Reply:
Just remove failed entirely and replace it with returj

Reply:
This was just a rough draft basically - I finialized sticky tapes today - I'll finish this tomorrow.